modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -202,3 +202,59 @@ START_TEST(PIDL)
         skip("?\n");
     ILFree(pidl);
 }
+
+START_TEST(ILIsEqual)
+{
+    LPITEMIDLIST p1, p2, pidl;
+
+    p1 = p2 = NULL;
+    ok_int(ILIsEqual(p1, p2), TRUE);
+
+    ITEMIDLIST emptyitem = {};
+    p1 = p2 = &emptyitem;
+    ok_int(ILIsEqual(p1, p2), TRUE);
+
+    ok_int(ILIsEqual(NULL, &emptyitem), FALSE); // These two are not equal for some reason
+
+    p1 = SHCloneSpecialIDList(NULL, CSIDL_DRIVES, FALSE);
+    p2 = SHCloneSpecialIDList(NULL, CSIDL_DRIVES, FALSE);
+    if (p1 && p2)
+    {
+        ok_int(ILIsEqual(p1, p2), TRUE);
+        p1->mkid.abID[0] = 0x2E; // Convert Desktop RegItem to Computer RegItem
+        ok_int(ILIsEqual(p1, p2), FALSE);
+    }
+    else
+    {
+        skip("?\n");
+    }
+    ILFree(p1);
+    ILFree(p2);
+
+    // ILIsParent must compare like ILIsEqual
+    p1 = SHSimpleIDListFromPath(L"c:\\");
+    p2 = SHSimpleIDListFromPath(L"c:\\dir\\file");
```

Comment:
Just to clarify, there is no obligation that these paths actually exist or don't exist on the system?

Reply:
They are virtual, yes.